core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java

```diff
@@ -386,6 +386,10 @@ public CHILD_CLASS json(final byte[]... jsonBytes) throws SchemaException {
         @Override
         @JsonIgnore
         public CHILD_CLASS merge(final View view) {
+            if (view instanceof NamedView && !(getThisView() instanceof NamedView)) {
```

Comment:
The View class should not reference the NamedView class. It causes a cyclic dependency between the 2 classes. Perhaps you could create a method:
```java
public boolean canBeMerged(final View destinationView, final View sourceView);
```

In named view you could override it to to check the destinationView is a NamedView.

Then in the merge method here you could do:
```java
this.canBeMerged(this, view) && view.canBeMerged(this, view);
```

That would check the View is happy for them to be merged and the NamedView class is happy.